### PR TITLE
8253470: Javadoc clean up in Filter and Headers

### PR DESCRIPTION
**Pull request by pconcannon**
Hi,

Could someone please review my doc-only fix for JDK-8253470 - 'Javadoc clean up in Filter and Headers' ?

This fix is set of formatting changes intended to clean up the javadoc of the following classes :

`com.sun.net.httpserver.Filter`
`com.sun.net.httpserver.Filter.Chain`
`com.sun.net.httpserver.Headers` 

This issue is a sub-task of [JDK-8252822](https://bugs.openjdk.java.net/browse/JDK-8252822)

Kind regards,
Patrick

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8253470](https://bugs.openjdk.java.net/browse/JDK-8253470): Javadoc clean up in Filter and Headers

### Reviewers
 * [Daniel Fuchs](https://openjdk.java.net/census#dfuchs) (@dfuch - **Reviewer**)
 * [Michael McMahon](https://openjdk.java.net/census#michaelm) (@Michael-Mc-Mahon - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/429/head:pull/429`
`$ git checkout pull/429`

### CODE REVIEW

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/Filter.java [hunk 1]
```diff
@@ -26,17 +26,21 @@
...
  * before the application's exchange handler is invoked, and post-processing
  * occurs after the exchange handler returns.  Filters
- * are organised in chains, and are associated with HttpContext instances.
- * <p>
- * Each Filter in the chain, invokes the next filter within its own
- * doFilter() implementation. The final Filter in the chain invokes the applications
- * exchange handler.
+ * are organised in chains, and are associated with {@link HttpContext} instances.
+ *
+ * <p> Each Filter in the chain, invokes the next filter within its own
```
Comment (dfuch) [inline src/jdk.httpserver/share/classes/com/sun/net/httpserver/Filter.java:40]: Here `Each Filter` should be either `Each filter` or `Each {@code Filter}`

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/Filter.java [hunk 2]
```diff
@@ -26,17 +26,21 @@
...
+ * {@link #doFilter(HttpExchange, Chain)} implementation. The final Filter in
```
Comment (dfuch) [inline src/jdk.httpserver/share/classes/com/sun/net/httpserver/Filter.java:41]: Same here for `the final Filter` - should be either `the final filter` or `the final {@code Filter}`

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/Filter.java [hunk 3]
```diff
@@ -95,35 +101,42 @@ public void doFilter (HttpExchange exchange) throws IOException {
...
      *
-     * @param exchange the {@code HttpExchange} to be filtered.
-     * @param chain the Chain which allows the next filter to be invoked.
-     * @throws IOException may be thrown by any filter module, and if
-     *          caught, must be rethrown again.
+     * <ul>
+     *     <li> Examine or modify the request headers.
+     *     <li> Filter the request body or the response body, by creating suitable
+     *     filter streams and calling {@link HttpExchange#setStreams(InputStream, OutputStream)}.
+     *     <li> Set attribute Objects in the exchange, which other filters or
```
Comment (dfuch) [inline src/jdk.httpserver/share/classes/com/sun/net/httpserver/Filter.java:109]: `Set attribute objects` ...

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/Filter.java [hunk 4]
```diff
@@ -47,39 +51,41 @@
...
+     * A chain of filters associated with a {@link HttpServer}.
+     * Each filter in the chain is given one of these so it can invoke the
+     * next filter in the chain.
      */
     public static class Chain {
-        /* the last element in the chain must invoke the users
-         * handler
+
+        /**
+         * The last element in the chain must invoke the users
```
Comment (Michael-Mc-Mahon) [inline src/jdk.httpserver/share/classes/com/sun/net/httpserver/Filter.java:61]: invoke the users -> invoke the user's

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/Headers.java [hunk 5]
```diff
@@ -36,43 +41,48 @@
...
- * <li>{@link #add(String,String)} adds the given header value to the list for the given key</li>
- * <li>{@link #set(String,String)} sets the given header field to the single value given
- * overwriting any existing values in the value list.
- * </ul><p>
- * All methods in this class accept <code>null</code> values for keys and values. However, null
+ *     <li>{@link #getFirst(String)} returns a single valued header or the first value of
+ *     a multi-valued header.
+ *     <li>{@link #add(String,String)} adds the given header value to the list for the given key
+ *     <li>{@link #set(String,String)} sets the given header field to the single value given
+ *     overwriting any existing values in the value list
```
Comment (dfuch) [inline src/jdk.httpserver/share/classes/com/sun/net/httpserver/Headers.java:62]: For consistency, either:
  - no item should end with a punctuation sign
  - or all of them should end with a coma, except the last that should end with a full stop.
  - or all of them should end with a full stop.

I am not sure what is the best, but it is weird that the first item ends with a full stop and the other two don't.

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/Headers.java [hunk 6]
```diff
@@ -36,43 +41,48 @@
...
  * one key "HeaderName" with two values "value1 and value2"
  * then this object is output as two header lines:
+ *
  * <blockquote><pre>
  * HeaderName: value1
  * HeaderName: value2
  * </pre></blockquote>
- * <p>
- * All the normal {@link java.util.Map} methods are provided, but the following
+ *
```
Comment (Michael-Mc-Mahon) [inline src/jdk.httpserver/share/classes/com/sun/net/httpserver/Headers.java:53]: The lines in the list look very long. Might be better to reformat to keep lines less than 80 chars (or whatever the limit is supposed to be).

**Review by dfuch — CHANGES_REQUESTED** [review]
Thanks for working on this issue Patrick! 

A few comments below:

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/Filter.java (hunk 4 above)
Comment (pconcannon) [reply src/jdk.httpserver/share/classes/com/sun/net/httpserver/Filter.java]: Change made in commit https://github.com/openjdk/jdk/pull/429/commits/4de7e6cac8681914f94c407b032fb8c1cb172828

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/Headers.java (hunk 6 above)
Comment (pconcannon) [reply src/jdk.httpserver/share/classes/com/sun/net/httpserver/Headers.java:53]: Change made in commit 4de7e6c

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/Filter.java (hunk 1 above)
Comment (pconcannon) [reply src/jdk.httpserver/share/classes/com/sun/net/httpserver/Filter.java]: Change made in commit 4de7e6c

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/Filter.java (hunk 2 above)
Comment (pconcannon) [reply src/jdk.httpserver/share/classes/com/sun/net/httpserver/Filter.java]: Change made in commit 4de7e6c - `the final Filter` --> `the final {@code Filter}`

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/Filter.java (hunk 3 above)
Comment (pconcannon) [reply src/jdk.httpserver/share/classes/com/sun/net/httpserver/Filter.java]: Change made in commit 4de7e6c

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/Headers.java (hunk 5 above)
Comment (pconcannon) [reply src/jdk.httpserver/share/classes/com/sun/net/httpserver/Headers.java]: Change made in commit 4de7e6c - all lines now end with a full stop

**Review by dfuch — APPROVED** [review]

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/Filter.java [hunk 7]
```diff
@@ -26,17 +26,21 @@
...
  * before the application's exchange handler is invoked, and post-processing
  * occurs after the exchange handler returns.  Filters
- * are organised in chains, and are associated with HttpContext instances.
- * <p>
- * Each Filter in the chain, invokes the next filter within its own
- * doFilter() implementation. The final Filter in the chain invokes the applications
- * exchange handler.
+ * are organised in chains, and are associated with {@link HttpContext} instances.
+ *
+ * <p> Each {@Filter} in the chain, invokes the next filter within its own
```
Comment (Michael-Mc-Mahon) [inline src/jdk.httpserver/share/classes/com/sun/net/httpserver/Filter.java:40]: \@Filter Is that supposed to be {@code Filter} or {@link Filter}?

**Review by Michael-Mc-Mahon — CHANGES_REQUESTED** [review]

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/Filter.java (hunk 7 above)
Comment (dfuch) [reply src/jdk.httpserver/share/classes/com/sun/net/httpserver/Filter.java:40]: Ah! I missed that. Well spotted Michael. Should be `{@code Filter}` of course.

**Review by dfuch — CHANGES_REQUESTED** [review]

**Review by dfuch — APPROVED** [review]

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/Filter.java (hunk 7 above)
Comment (pconcannon) [reply src/jdk.httpserver/share/classes/com/sun/net/httpserver/Filter.java]: Thanks for spotting this, Michael. I've rectified this in the latest commit (https://github.com/openjdk/jdk/pull/429/commits/c4a654832e81740543313e2a86a245865f3283b9).

**Review by Michael-Mc-Mahon — APPROVED** [review]

**Pull request closed without merging**